bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WindowsDefenderConfigurator.java

```diff
@@ -0,0 +1,351 @@
+
+					Button performExclusion = radioButtonFactory
+							.text(bindProductName(WorkbenchMessages.WindowsDefenderConfigurator_performExclusionChoice))
+							.onSelect(e -> {
+								choice[0] = HandlingOption.EXECUTE_EXCLUSION;
+								getButton(0).setEnabled(true);
+							}).create(parent);
+					Point size = performExclusion.computeSize(SWT.DEFAULT, SWT.DEFAULT);
+					GridDataFactory.swtDefaults().hint((int) (size.x * 0.6), (int) (size.y * 2.3)).indent(0, 5)
+							.applyTo(performExclusion);
+
+					Button keepScanning = radioButtonFactory
+							.text(bindProductName(
+									WorkbenchMessages.WindowsDefenderConfigurator_ignoreThisInstallationChoice))
+							.onSelect(e -> {
+								choice[0] = HandlingOption.IGNORE_THIS_INSTALLATION;
+								getButton(0).setEnabled(true);
+							}).create(parent);
+					GridDataFactory.swtDefaults().indent(0, 5).applyTo(keepScanning);
+
+					LinkFactory.newLink(SWT.WRAP)
+							.text(WorkbenchMessages.WindowsDefenderConfigurator_detailsAndOptionsLinkText)
+							.onSelect((e -> {
+								PreferenceDialog dialog = PreferencesUtil.createPreferenceDialogOn(null,
+										"org.eclipse.ui.preferencePages.Startup", null, null); //$NON-NLS-1$
+								dialog.setBlockOnOpen(false);
+								dialog.open();
+								this.setReturnCode(Window.CANCEL);
+								this.close();
+							})).create(parent);
+
+					return parent;
+				}
+
+				@Override
+				protected void createButtonsForButtonBar(Composite parent) {
+					super.createButtonsForButtonBar(parent);
+					getButton(0).setEnabled(false);
+					getButton(1).forceFocus(); // prevents auto-focusing one of the radio-buttons which selects them
+				}
+			};
+			int open = dialog.open();
+			return open == Window.OK ? choice[0] : null;
+		});
+	}
+
+	public static String bindProductName(String message) {
+		return NLS.bind(message, Platform.getProduct().getName());
+	}
+
+	public static IEclipsePreferences getPreference(IScopeContext instance) {
+		return instance.getNode(PI_WORKBENCH);
+	}
+
+	public static void savePreference(IScopeContext scope, String key, String value) throws CoreException {
+		IEclipsePreferences preferences = getPreference(scope);
+		preferences.put(key, value);
+		try {
+			preferences.flush();
+		} catch (BackingStoreException e) {
+			throw new CoreException(Status.error("Failed to safe preference " + preferences, e)); //$NON-NLS-1$
+		}
+	}
+
+	private static Optional<Path> getInstallationLocation() {
+		// For install location see:
+		// https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fmisc%2Fruntime-options.html&anchor=locations
+		try {
+			Location installLocation = Platform.getConfigurationLocation();
+			return Optional.of(Path.of(installLocation.getURL().toURI())); // assume location has a file-URL
+		} catch (URISyntaxException e) { // ignore
+		}
+		return Optional.empty();
+	}
+
+	private static List<Path> getExecutablePath() {
+		// TODO: Alternatively use
```

Comment:
what about this TODO?

Reply:
Removed the commented out code. The currently used one, should be sufficient.